### PR TITLE
Primary keys now compatible with primary keys from the Object Store.

### LINKED ISSUE #1703: realm file incompatible if model has PrimaryKey
from slack.

Realm Swift reports an error when opening the realm file generated on Android.

model definition in Java:

```
public class DataItem extends RealmObject {
    @PrimaryKey
    private int id;

    // seter and getter here
}
```

model definition in Swift:

```
class DataItem: Object {
    dynamic var id = 0

    override static func primaryKey() -> String? {
        return "id"
    }
}
```

reported error:

```
fatal error: 'try!' expression unexpectedly raised an error: Error Domain=io.realm Code=0 "Migration is required for object type 'DataItem' due to the following errors:
- Property 'id' has been made a primary key." UserInfo={NSLocalizedDescription=Migration is required for object type 'DataItem' due to the following errors:
- Property 'id' has been made a primary key.}: file /Library/Caches/com.apple.xbs/Sources/swiftlang_PONDEROSA/swiftlang_PONDEROSA-700.1.101.6/src/swift/stdlib/public/core/ErrorType.swift, line 50
```

### PR DESCRIPTION
**Pull request by cmelchior**
Fixes #1703 

@realm/java 

This PR doesn't try to unify where `Table.TABLE_PREFIX` is set (see also #766) . Would adding a constant to e.g `Utils.hpp` be an idea. Then we could load statically when the Table class is loaded the first time through some `getTablePrefix()` method?

### CODE REVIEW

**Comment by zaki50** [comment]
retest this please

File: examples/introExample/src/main/java/io/realm/examples/intro/model/AnnotationTypes.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2014 Realm Inc.
```
Comment (kneth) [inline examples/introExample/src/main/java/io/realm/examples/intro/model/AnnotationTypes.java]: "2014" -> "2015"

**Comment by kneth** [comment]
:+1:

File: realm/realm-jni/src/io_realm_internal_table.cpp [hunk 2]
```diff
@@ -1588,7 +1588,7 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_Table_nativeSetPrimaryKey(
     try {
         Table* table = TBL(nativeTablePtr);
         Table* pk_table = TBL(nativePrivateKeyTablePtr);
-        const char* table_name = table->get_name().data();
+        const char* table_name = table->get_name().substr(6).data(); // Remove "class_" prefix
```
Comment (zaki50) [inline realm/realm-jni/src/io_realm_internal_table.cpp]: It's no longer table name.
Should be "model_name"?

Comment (cmelchior) [inline realm/realm-jni/src/io_realm_internal_table.cpp]: I believe we still uses Cores terminology in the internal API's, e.g. Table/Column/Rows.

**Comment by dalinaum** [comment]
:+1:

File: changelog.txt [hunk 3]
```diff
@@ -8,7 +9,7 @@
  * Updated ProGuard configuration. See documentation https://realm.io/docs/java/latest/#proguard for more details.
  * Updated Kotlin example to use 1.0.0-beta.
  * Fixed warnings reported by `lint -Xlint:all` (#1644).
- * Fixed a bug where simultaneous opening and closing a Realm on different threads might result in a NullPointerException (#1646).
+ * Fixed a bug where simultaneous opening and closing a Realm on diffesrent threads might result in a NullPointerException (#1646).
```
Comment (jpsim) [inline changelog.txt]: you introduced a typo here

**Pull request merged**